jenkins/jobs/bmo_e2e_optional_tests.pipeline

```diff
@@ -0,0 +1,89 @@
+import java.text.SimpleDateFormat
+
+// 2 hour
+def TIMEOUT = 7200
+
+// Set defaults for non-PR jobs
+def pullSha = (env.PULL_PULL_SHA) ?: "main"
+def pullBase = (env.PULL_BASE_REF) ?: "main"
+def repoUrl = "https://github.com/metal3-io/baremetal-operator.git"
+// Fetch the base branch and the pullSha, nothing else
+def refspec = '+refs/heads/' + pullBase + ':refs/remotes/origin/' + pullBase + ' ' + pullSha
+
+pipeline {
+  environment {
+    GINKGO_FOCUS="${GINKGO_FOCUS}"
+  }
+  agent none
+  stages {
+    stage("Run Baremetal Operator optional e2e tests") {
+      matrix {
+        agent { label "metal3ci-8c16gb-ubuntu" }
+        // Skip redfish on PRs, test all for periodic jobs
+        when {
+          anyOf {
+            expression { env.BMC_PROTOCOL != "redfish" }
+            triggeredBy 'TimerTrigger'
+          }
+        }
```

Comment:
This is maybe a bit unnecessary for the optional test. We do it for the mandatory tests to conserve some resources without much risk, but these will anyway not run unless explicitly triggered.
Not gonna block for that though. We can change it later if needed.